cmd/serve/serve.go

```diff
@@ -56,6 +80,36 @@ func NewServeCmd() *cobra.Command {
 			shutdown := make(chan os.Signal, 1)
 			signal.Notify(shutdown, syscall.SIGINT, syscall.SIGTERM)
 
+			// -------------------------------------------------------------------------
+			// Start Debug Service
+			router := gin.Default()
+			router.Use(gin.Recovery())
+			router.Use(sloggin.New(logger.L.WithGroup("debug")))
+
+			// register pprof middleware endpoints
+			pprof.Register(router)
+
+			// register expvar endpoints
+			router.GET("/debug/vars", expvar.Handler())
+
+			go func() {
+				logger.L.InfoContext(ctx, "startup", "status", "debug v1 router started", "host", debugOpts.Addr)
+
+				srv := http.Server{
+					Addr:         debugOpts.Addr,
+					Handler:      router,
+					ReadTimeout:  debugOpts.ReadTimeout,
+					IdleTimeout:  debugOpts.IdleTimeout,
+					WriteTimeout: debugOpts.WriteTimeout,
+				}
```

Comment:
I thought we were supposed to expose the `debug` and `health` endpoint within the same server, having another endpoint is fine, just wonder if something has changed in the meantime

Reply:
I had the same conversation with Rinaldo he mentioned he prefer two separate endpoints. To be honest, I think, like you do, that one is sufficient.

Reply:
No strong opinion here, if you think too that one endpoint is enough, let's go for it but.